src/setupWorker/start/utils/getWorkerByRegistration.ts

```diff
@@ -1,9 +1,20 @@
 /**
- * Attempts to resolve a Service Worker instance from any of its states:
- * active, installing, or waiting.
+ * Attempts to resolve a Service Worker instance from a given registration,
+ * regardless of its state (active, installing, waiting).
  */
 export const getWorkerByRegistration = (
   registration: ServiceWorkerRegistration,
+  absoluteWorkerUrl: string,
 ): ServiceWorker | null => {
-  return registration.active || registration.installing || registration.waiting
+  const allStates = [
+    registration.active,
+    registration.installing,
+    registration.waiting,
+  ]
+  const existingStates = allStates.filter(Boolean) as ServiceWorker[]
```

Comment:
I'm filtering the possible states of the registration so when it comes to finding a relevant one, we're always operating on existing registrations.